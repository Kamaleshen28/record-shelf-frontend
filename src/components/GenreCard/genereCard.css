
.genre-all-song-container{
    display: flex;
    flex-direction: column;
    gap:35px
}

.genre-title-container{
    display: flex;
    align-items: center;
}

.genre-image{
    width:110px;
}

.genre-name{
    position: relative;
    left:-40px;
    background-color: #4C63AA;
    min-width:160px ;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    color: white;
    font-size: 20px;
    font-weight: bolder;
    border-radius: 20px;
}

.all-songs-container{
    display: flex;
    flex-wrap: wrap;
    gap:10px;
}