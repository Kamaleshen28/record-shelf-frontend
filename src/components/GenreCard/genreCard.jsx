import React from 'react'
import Card from '../Card/card'
import './genereCard.css'

export default function GenreCard({songs}){
    
    const genreName = (songs[0].genre.name).toLowerCase();
    const renderAllSongs = songs.map((song) => {
        return (<Card key={song.id} {...song}/>)
    })

    return(
        <div className='genre-all-song-container'>
            <div className='genre-title-container'>
                <img className='genre-image' src={require(`../../assets/genre-${genreName}.png`)} alt='Not Found' />
                <a className='genre-name' >{genreName}</a>
            </div>
            <div className='all-songs-container'>
                {renderAllSongs}
            </div>
        </div>
    )
}