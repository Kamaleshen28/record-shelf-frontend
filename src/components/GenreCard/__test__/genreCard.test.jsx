import { render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import GenreCard from '../genreCard';


const mockSongData =[ 
    {
        'id': '0e2ddccf-f494-4c41-a499-90f8267f491a',
        'name': 'Beautiful Mistakes (feat. Megan Thee Stallion)',
        'genre': {
            'id': '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
            'name': 'Pop'
        },
        'artist': {
            'id': '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
            'name': 'Maroon 5'
        },
        'imageUrl': 'https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631',
        'publishedAt': '2021-06-11T00:00:00'
    },
    {
        'id': '45e1d753-2986-43cb-9b9d-30c370056319',
        'name': 'This Love',
        'genre': {
            'id': '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
            'name': 'Pop'
        },
        'artist': {
            'id': '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
            'name': 'Maroon 5'
        },
        'imageUrl': 'https://i.scdn.co/image/ab67616d0000b27317b3850d758fff5a2301e537',
        'publishedAt': '2002-06-25T00:00:00'
    }
];




describe('GenreCard', () => {
    it('should render genreCard correctly', async () => {
        const {asFragment} = render(<GenreCard songs={mockSongData}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    
});