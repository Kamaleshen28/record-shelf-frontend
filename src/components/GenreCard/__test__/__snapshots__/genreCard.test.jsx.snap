// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreCard should render genreCard correctly 1`] = `
<DocumentFragment>
  <div
    class="genre-all-song-container"
  >
    <div
      class="genre-title-container"
    >
      <img
        alt="Not Found"
        class="genre-image"
        src="genre-pop.png"
      />
      <a
        class="genre-name"
      >
        pop
      </a>
    </div>
    <div
      class="all-songs-container"
    >
      <div
        class="card dark-card"
      >
        <div
          class="card-image-container"
        >
          <img
            alt="Not Found"
            class="card-image"
            src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
          />
        </div>
        <div
          class="details-container"
        >
          <div
            class="text-container"
          >
            <a
              class="song-name-text"
            >
              Beautiful Mistakes (feat. Megan Thee Stallion)
            </a>
            <a
              class="song-genre-text"
            >
              Pop
            </a>
          </div>
          <div
            class="heart-image-container"
          >
            <img
              alt="Not Found"
              class="heart-image"
              data-testid="heart-image"
              src="heart-gray.svg"
            />
            <a
              class="count-text"
              data-testid="like-count"
            />
          </div>
        </div>
      </div>
      <div
        class="card dark-card"
      >
        <div
          class="card-image-container"
        >
          <img
            alt="Not Found"
            class="card-image"
            src="https://i.scdn.co/image/ab67616d0000b27317b3850d758fff5a2301e537"
          />
        </div>
        <div
          class="details-container"
        >
          <div
            class="text-container"
          >
            <a
              class="song-name-text"
            >
              This Love
            </a>
            <a
              class="song-genre-text"
            >
              Pop
            </a>
          </div>
          <div
            class="heart-image-container"
          >
            <img
              alt="Not Found"
              class="heart-image"
              data-testid="heart-image"
              src="heart-gray.svg"
            />
            <a
              class="count-text"
              data-testid="like-count"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
